cloud/scope/powervs_cluster_test.go

```diff
@@ -3043,3 +3045,728 @@ func TestCreateVPCSubnet(t *testing.T) {
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockTransitGateway = mockTG.NewMockTransitGateway(mockCtrl)
+		mockVPC = mock.NewMockVpc(mockCtrl)
+		mockResourceController = mockRC.NewMockResourceController(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("when TransitGatewayID is set in Status, return error getting TransitGateway", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			TransitGatewayClient: mockTransitGateway,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					TransitGateway: &infrav1beta2.TransitGatewayStatus{
+						ID: ptr.To("transitGatewayID"),
+					},
+				},
+			},
+		}
+
+		mockTransitGateway.EXPECT().GetTransitGateway(gomock.Any()).Return(nil, nil, errors.New("failed to get transit gateway"))
+		requeue, err := clusterScope.ReconcileTransitGateway()
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("When TransitGateway already exists, returns error when getting TransitGateway connections", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			TransitGatewayClient: mockTransitGateway,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					TransitGateway: &infrav1beta2.TransitGatewayStatus{
+						ID: ptr.To("transitGatewayID"),
+					},
+				},
+			},
+		}
+
+		mockTransitGateway.EXPECT().GetTransitGateway(gomock.Any()).Return(&tgapiv1.TransitGateway{Name: ptr.To("transitGatewayName"), Status: ptr.To(string(infrav1beta2.TransitGatewayStateAvailable))}, nil, nil)
+		mockTransitGateway.EXPECT().ListTransitGatewayConnections(gomock.Any()).Return(nil, nil, errors.New("failed to get transitGateway connections"))
+		requeue, err := clusterScope.ReconcileTransitGateway()
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("When TransitGatewayID is set in Status and TransitGateway not in available state", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			TransitGatewayClient: mockTransitGateway,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					TransitGateway: &infrav1beta2.TransitGatewayStatus{
+						ID: ptr.To("transitGatewayID"),
+					},
+				},
+			},
+		}
+
+		mockTransitGateway.EXPECT().GetTransitGateway(gomock.Any()).Return(&tgapiv1.TransitGateway{Name: ptr.To("transitGatewayName"), Status: ptr.To(string(infrav1beta2.TransitGatewayStatePending))}, nil, nil)
+		requeue, err := clusterScope.ReconcileTransitGateway()
+		g.Expect(requeue).To(BeTrue())
+		g.Expect(err).To(BeNil())
+	})
+
+	t.Run("When TransitGatewayID is set in Spec, returns error while getting TransitGateway details", func(t *testing.T) {
```

Comment:
How about covering the case where TG with spec.ID or name already exists in cloud and is in available state?

Reply:
added new UT for this